src/main/java/com/redhat/rhjmc/containerjfr/platform/openshift/OpenShiftPlatformClient.java

```diff
@@ -99,14 +100,21 @@ private boolean isCompatiblePort(EndpointPort port) {
         return "jfr-jmx".equals(port.getName()) || 9091 == port.getPort();
     }
 
-    private List<ServiceRef> createServiceRefs(EndpointSubset subset, EndpointPort port) {
+    private List<ServiceRef> createServiceRefs(
+            Logger logger, EndpointSubset subset, EndpointPort port) {
         return subset.getAddresses().stream()
                 .map(
-                        addr ->
-                                new ServiceRef(
+                        addr -> {
+                            try {
+                                return new ServiceRef(
                                         addr.getIp(),
-                                        addr.getTargetRef().getName(),
-                                        port.getPort()))
+                                        port.getPort(),
+                                        addr.getTargetRef().getName());
+                            } catch (Exception e) {
+                                logger.warn(e);
+                                return null;
```

Comment:
Since there is a possible `return null` in this chain, there should also be a corresponding filter to remove null entries, like the `KubeEnv` client has for example.

Reply:
Fixed.